channel/src/main/java/com/synopsys/integration/alert/channel/jira2/server/delegate/JiraServerIssueCreator.java

```diff
@@ -0,0 +1,115 @@
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.persistence.model.job.details.JiraServerJobDetailsModel;
+import com.synopsys.integration.alert.descriptor.api.JiraServerChannelKey;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.model.components.ProjectComponent;
+import com.synopsys.integration.jira.common.model.request.builder.IssueRequestModelFieldsMapBuilder;
+import com.synopsys.integration.jira.common.model.response.IssueCreationResponseModel;
+import com.synopsys.integration.jira.common.model.response.IssueResponseModel;
+import com.synopsys.integration.jira.common.server.model.IssueCreationRequestModel;
+import com.synopsys.integration.jira.common.server.service.IssueService;
+import com.synopsys.integration.jira.common.server.service.ProjectService;
+
+public class JiraServerIssueCreator extends JiraIssueCreator<IssueCreationRequestModel> {
+    private final JiraServerJobDetailsModel distributionDetails;
+    private final IssueService issueService;
+    private final ProjectService projectService;
+    private final JiraIssueCreationRequestCreator jiraIssueCreationRequestCreator;
+
+    public JiraServerIssueCreator(
+        JiraServerChannelKey jiraServerChannelKey,
+        JiraServerIssueCommenter jiraServerIssueCommenter,
+        IssueTrackerCallbackInfoCreator callbackInfoCreator,
+        JiraServerJobDetailsModel distributionDetails,
+        IssueService issueService,
+        ProjectService projectService,
+        JiraIssueCreationRequestCreator jiraIssueCreationRequestCreator,
+        JiraIssueAlertPropertiesManager issuePropertiesManager,
+        JiraErrorMessageUtility jiraErrorMessageUtility
+    ) {
+        super(
+            jiraServerChannelKey,
+            jiraServerIssueCommenter,
+            callbackInfoCreator,
+            jiraErrorMessageUtility,
+            issuePropertiesManager,
+            JiraServerDescriptor.KEY_ISSUE_CREATOR
+        );
+        this.distributionDetails = distributionDetails;
+        this.issueService = issueService;
+        this.projectService = projectService;
+        this.jiraIssueCreationRequestCreator = jiraIssueCreationRequestCreator;
+    }
+
+    @Override
+    public IssueCreationRequestModel createIssueCreationRequest(IssueCreationModel alertIssueCreationModel) throws AlertException {
+        ProjectComponent jiraProject = retrieveProjectComponent();
+        JiraCustomFieldReplacementValues replacementValues = alertIssueCreationModel.getSource()
+                                                                 .map(this::createCustomFieldReplacementValues)
+                                                                 .orElse(JiraCustomFieldReplacementValues.trivial(alertIssueCreationModel.getProvider()));
+        IssueRequestModelFieldsMapBuilder fieldsBuilder = jiraIssueCreationRequestCreator.createIssueRequestModel(
```

Comment:
It seems like this class is common. Any chance you could make this the argument of this method and move the above logic into the superclass?